source/js/text-ellipsis.js

```diff
@@ -1,14 +1,92 @@
-/*** 
- * @summary Sets the ellipsis based on the cpl - 3.
- * @param {Object} textLine: Node element, where to set the ellipsis
- * @param {Integer} characterPerLine: Characters per line. Set or calculate cpl.
+var cpl = require("./characters-per-line");
+var ellipsis = require("html-ellipsis");
+cpl.initFontConstant();
+
+var EllipsisUtil = {
+  /***
+   * @param {Object} lineStyles: Node with DOM elements that has a string
+   * @return {Boolean}
+   */
+  hasSpecificHeight: function(lineStyles) {
+    return lineStyles.height !== "auto" && parseInt(lineStyles.height) !== 0;
+  },
+
+  /***
+   * @summary Returns the width of the object
+   * @param {Node} textLines: Node with DOM elements that has a string
+   * @return {Integer} width
+   */
+  getWidth: function(textLine) {
+    var lineStyles = window.getComputedStyle(textLine);
+
+    if(EllipsisUtil.hasSpecificHeight(lineStyles)) {
+      return parseInt(lineStyles.width);
+    }
+  },
+
+  /***
+   * @param {Node} element Node element.
+   * @return {Integer} Returns a font size.
+   * */
+  getFontSize: function(element) {
+    return parseInt(window.getComputedStyle(element).getPropertyValue("font-size"));
+  },
+
+  /***
+   * @summary Returns the width of the object's parent
+   * @param {Node} textLines: Node with DOM elements that has a string
+   * @return {Integer} parent's width
+   */
+  getParentsWidth: function(textLine) {
+    var lineStyles = window.getComputedStyle(textLine);
+
+    if (EllipsisUtil.hasSpecificHeight(lineStyles)) {
+      var parentElement = textLine.parentElement;
+      var parentStyles = window.getComputedStyle(parentElement);
+
+      if (parentElement.className !== "multi-teaser-hotel-name") {
+        return parseInt(parentStyles.width);
+      }
+
+      var hotelRating = parentElement.querySelector(".mot_stars");
```

Comment:
ErrorHandling is good to use here. If mod-stars is not being found, the whole bundle.js won't work.
At least it was the case, when one of our scripts didn't find a specific element, which was not related to the navigation, but the navigation didn't work.